source/s3_client.c

```diff
@@ -25,24 +25,33 @@
 #include <inttypes.h>
 #include <math.h>
 
+enum aws_s3_meta_request_work_op {
+    AWS_S3_META_REQUEST_WORK_OP_PUSH,
+    AWS_S3_META_REQUEST_WORK_OP_REMOVE,
+};
+
+struct aws_s3_meta_request_work {
+    struct aws_linked_list_node node;
+    struct aws_s3_meta_request *meta_request;
+    enum aws_s3_meta_request_work_op op;
+};
+
 static const uint32_t s_s3_max_request_count_per_connection = 100;
+static const uint32_t s_connection_timeout_ms = 3000;
+
+/* TODO Provide analysis on origins of this value. */
+static const double s_throughput_per_vip_gbps = 6.25;
+static const uint32_t s_num_connections_per_vip = 10;
 
 static const uint16_t s_http_port = 80;
 static const uint16_t s_https_port = 443;
 
-static const uint64_t s_default_part_size = 20 * 1024 * 1024;
+static const uint64_t s_default_part_size = 5 * 1024 * 1024;
```

Comment:
Should also provide info on these values too

Reply:
Added a TODO for now.